src/influenzanet-verdi-contact-extension/studyRules.ts

```diff
@@ -167,48 +167,57 @@ export const reassignContactsSurvey = () =>
     )
   );
 
+const temporaryFlagKeyForContactsSurveyStart = 'lastContactsSurveyStart';
+
+const saveLastContactsSurveyStartAsFlag = () => StudyEngine.if(
+  StudyEngine.participantState.hasSurveyKeyAssigned(surveyKeys.Contacts),
+  StudyEngine.participantActions.updateFlag(
+    temporaryFlagKeyForContactsSurveyStart,
+    StudyEngine.participantState.getSurveyKeyAssignedFrom(surveyKeys.Contacts),
+  ),
+  // handle if interval survey is not assigned yet
+  // Assume it was 4 weeks ago:
+  StudyEngine.participantActions.updateFlag(
+    temporaryFlagKeyForContactsSurveyStart,
+    StudyEngine.timestampWithOffset({ days: -28 }),
```

Comment:
Probably this isn't necessary... the "enter" rules take care of the entry situation. 
Right now, I am unsure if this should ever happen (that the survey is unexpectedly not assigned, but the thought, ok, let's be explicit about it.. 
Yes, I wanted to add zero offset first, but then it might happen again, that we are in the quarter we want to add. (In the final else case of the assignments, if everything fails for quarter selection) 

Hm... since this should never happen, I am flexibel changing the value - you prefer zero offset? (Is maybe less confusing) my other instinct would have been to subtract 91 days, but seems also unnecessary.

Reply:
Yeah, I think using a zero offset is the best thing to do here since this would reproduce the same behavior this had previously and we have no strong reasons to change that.